src/cbc_sdk/platform/jobs.py

```diff
@@ -84,6 +85,45 @@ def get_progress(self):
         self._info['progress'] = resp
         return resp['num_total'], resp['num_completed'], resp.get('message', None)
 
+    def _await_completion(self):
+        """
+        Waits for this job to complete by examining the progress data.
+
+        Required Permissions:
+            jobs.status(READ)
+
+        Returns:
+            Job: This object.
+        """
+        progress_data = (1, 0, '')
+        sleep_time = 0.0
+        while progress_data[1] < progress_data[0]:
+            if sleep_time > 0.0:
+                time.sleep(sleep_time)
+            try:
+                progress_data = self.get_progress()
+            except ServerError:
+                progress_data = (1, 0, '')
+            if sleep_time > 0.0:
+                if sleep_time < 1.0:
+                    sleep_time *= 2.0
+            else:
+                sleep_time = 0.1
+        return self
+
+    def await_completion(self):
+        """
+        Waits for this job to complete in an asynchronous fashion.
```

Comment:
I would clarify this description that the function returns a future which provides a non blocking mechanism to fetch the results when they are ready. This function does not block

Reply:
"Waits for this job to complete in an asynchronous fashion.  Returns a Future object which can be used to await results that are ready to fetch. This function call does not block."

Reply:
This phrasing to me is misleading `Waits for this job to complete `. How about this instead? `Create a python future to check for job completion and return results when available`

Reply:
"Python" and "Future" need to be capitalized, but otherwise I'll use that.